From 20f39c4aef3e438e8f986afcf28a70f05f042aee Mon Sep 17 00:00:00 2001
From: Vicentiu Galanopulo <vicentiu@balena.io>
Date: Tue, 28 Jul 2020 15:46:54 +0200
Subject: [PATCH] Update gpios pins to reflect etcher-pro hardware changes in v2_3

 TFT_RST (GPIO3_IO25, pin142) removed
 TFT_SHLR (GPIO4_IO1, pin192) removed
 TFT_UPDN (GPIO4_IO16, pin194) removed
 TouP_RTS (GPIO4_IO23, pin203) removed
 EE_WP (GPIO3_IO18, pin81) removed
 CEC_STBY (GPIO5_IO29, pin3) removed
 UART1 (GPIO_5_IO22, GPIO_5_IO23, pin69, pin 65) removed
 All HDMI signals removed
 LED_RST (GPIO5_IO4, pin198) moved on (GPIO4_IO23,pin203)
 PWM_FAN (GPIO5_IO4) added. Should be configured as PWM out for the FAN Control.
                   For the time can be set it just as a simple High Level output.
 uPCIe_1_RST (GPIO4_IO7, pin139) added. Set it as output and High Level on startup
 SD_WP (GPIO2_IO20, pin67) added.
 INT_IO_EXP (GPIO5_IO28) added
 TFT_BCL_PWM (GPIO5_IO18, pin43), Should be configured as PWM out
                              for the backlight Control.
                       For the time can be set it just as a simple High Level output.


Upstream-status: Inappropriate [configuration]
Signed-off-by: Vicentiu Galanopulo <vicentiu@balena.io>
---
 board/compulab/cl-som-imx8/cl-som-imx8.c | 13 +++++--------
 1 file changed, 5 insertions(+), 8 deletions(-)

diff --git a/board/compulab/cl-som-imx8/cl-som-imx8.c b/board/compulab/cl-som-imx8/cl-som-imx8.c
index 2a14ceb959..2d881d0059 100644
--- a/board/compulab/cl-som-imx8/cl-som-imx8.c
+++ b/board/compulab/cl-som-imx8/cl-som-imx8.c
@@ -336,16 +336,13 @@ int board_init(void)
 	gpio_direction_output(IMX_GPIO_NR(4, 5), 1);
 
 	gpio_request(IMX_GPIO_NR(5, 18),"tft_bcl_pwm");
-	gpio_direction_output(IMX_GPIO_NR(5, 18), 0);
+	gpio_direction_output(IMX_GPIO_NR(5, 18), 1);
 
-	gpio_request(IMX_GPIO_NR(4, 1),"tft_shlr");
-	gpio_direction_output(IMX_GPIO_NR(4, 1), 0);
+	gpio_request(IMX_GPIO_NR(5, 4),"pwm_fan");
+	gpio_direction_output(IMX_GPIO_NR(5, 4), 1);
 
-	gpio_request(IMX_GPIO_NR(4, 16),"tft_updw");
-	gpio_direction_output(IMX_GPIO_NR(4, 16), 1);
-
-	gpio_request(IMX_GPIO_NR(3, 25),"tft_rst");
-	gpio_direction_output(IMX_GPIO_NR(3, 25), 1);
+	gpio_request(IMX_GPIO_NR(4, 7),"uPCIe_1_RST");
+	gpio_direction_output(IMX_GPIO_NR(4, 7), 1);
 
 #if defined(CONFIG_USB_DWC3) || defined(CONFIG_USB_XHCI_IMX8M)
 	init_usb_clk();
-- 
2.17.1
